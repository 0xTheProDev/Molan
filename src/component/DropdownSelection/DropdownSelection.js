// @flow
import React from 'react';
import { Select } from 'semantic-ui-react';

import template from 'util/template';

const langOptions = Array.from(template, e => {
  return {
    key: e.lang,
    value: e.lang,
    text: e.name
  };
});

type DropdownSelectionProps = {
  defaultLang: string,
  onChange: (key: string, value: string) => void,
};

export function DropdownSelection(props: DropdownSelectionProps) {
  return (
    <Select
      placeholder="Select a language"
      value={props.defaultLang}
      options={langOptions}
      onChange={props.onChange}
    />
  );
}
