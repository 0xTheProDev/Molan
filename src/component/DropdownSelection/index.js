import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Select } from 'semantic-ui-react';

const langOptions = [
    { key: 1, value: "c", text: "C" }
];

export default class DropdownSelection extends Component {
    static propTypes = {
        defaultLang: PropTypes.string.isRequired,
        onChange:    PropTypes.func.isRequired
    };

    render() {
        return (
            <Select
              placeholder="Select a language"
              value={this.props.defaultLang}
              options={langOptions}
              onChange={this.props.onChange}
            />
        );
    }
}