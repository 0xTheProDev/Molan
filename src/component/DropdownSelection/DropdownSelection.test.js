import React from 'react';
import { Select } from 'semantic-ui-react';

import { DropdownSelection } from './DropdownSelection';

describe('Dropdown Selection', () => {
  let
    onChangeStub,
    minProps;

  beforeAll(() => {
    onChangeStub = jest.fn();

    minProps = {
      defaultLang: 'c',
      onChange: onChangeStub,
    };
  });

  it('should render', () => {
    const wrapper = shallow(<DropdownSelection {...minProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should change language on select', () => {
    const wrapper = shallow(<DropdownSelection {...minProps} />);
    wrapper.find(Select).simulate('change');
    expect(onChangeStub).toHaveBeenCalled();
  });
});
