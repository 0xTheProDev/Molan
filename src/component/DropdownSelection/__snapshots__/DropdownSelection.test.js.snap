// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Selection should render 1`] = `
<Select
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "key": "c",
        "text": "C (GCC 6.3)",
        "value": "c",
      },
      Object {
        "key": "cpp",
        "text": "C++ (G++ 6.3)",
        "value": "cpp",
      },
      Object {
        "key": "java",
        "text": "Java (OpenJDK 0.8)",
        "value": "java",
      },
      Object {
        "key": "python",
        "text": "Python (Python 2.7)",
        "value": "python",
      },
      Object {
        "key": "python3",
        "text": "Python (Python 3.6)",
        "value": "python3",
      },
      Object {
        "key": "MIPS",
        "text": "MIPS-IV (Simulator 0.4)",
        "value": "MIPS",
      },
      Object {
        "key": "Verilog",
        "text": "Verilog (iVerilog 10.0)",
        "value": "Verilog",
      },
    ]
  }
  placeholder="Select a language"
  value="c"
/>
`;
